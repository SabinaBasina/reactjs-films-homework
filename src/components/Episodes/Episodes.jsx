/* eslint-disable react/prop-types */
/* eslint-disable no-underscore-dangle */
import React, { Component } from 'react';
import http from 'axios';
import Episode from '../Episode/Episode';

class Episodes extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tvShow: [],
    };

    const { nameTvShow } = this.props;

    http
      .get(`https://api.tvmaze.com/singlesearch/shows?q=${nameTvShow}&embed=episodes`)
      .then((response) => {
        this.setState({ tvShow: response.data });
      });
  }

  render() {
    const { tvShow } = this.state;
    return (
      <div>
        { tvShow._embedded
            && tvShow._embedded.episodes.map(episode => (
              <Episode data={episode} />
            ))}
      </div>
    );
  }
}

export default Episodes;
