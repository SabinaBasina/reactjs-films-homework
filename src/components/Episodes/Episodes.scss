@import '../../pages/TvShowDetails/TvShowDetails.scss';
