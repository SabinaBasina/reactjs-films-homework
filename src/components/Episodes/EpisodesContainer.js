import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import Episodes from './Episodes';
import { loadTvShowEpisodes } from '../../redux/actions/episodes';
import { getTvShowEpisodes } from '../../redux/selectors';

const mapStateToProps = createStructuredSelector({
  tvShowEpisodes: getTvShowEpisodes,
});

const mapDispatchToProps = {
  loadTvShowEpisodes,
};

export default connect(mapStateToProps, mapDispatchToProps)(Episodes);
