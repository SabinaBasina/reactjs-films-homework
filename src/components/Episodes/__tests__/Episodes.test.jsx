/* eslint-disable no-shadow */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Renderer from 'react-test-renderer';
import http from 'axios';
import MockAdapter from 'axios-mock-adapter';
import Episodes from '../Episodes';

/* global test, expect, jest */

jest.mock(
  '../../Episode/Episode',
  () => data => `mock data${JSON.stringify(data)}`,
);

test.skip('Episodes test', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Episodes />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

test('Episodes test 2', () => {
  const mock = new MockAdapter(http);
  const tvShow = {
    _embedded: {
      episodes: [{ data: 1 }, { data: 2 }],
    },
  };
  mock.onGet('https://api.tvmaze.com/singlesearch/shows?q=nameTvShow&embed=episodes')
    .reply(200, tvShow);

  const renderer = Renderer.create(<Episodes nameTvShow="nameTvShow" />);
  const testInstance = renderer.toJSON();
  expect(testInstance).toMatchSnapshot();
  // const res = testInstance.findByProps({class: 'episode'}).length;
  // //const res = testInstance.children.length;
  // console.log(res);
});
