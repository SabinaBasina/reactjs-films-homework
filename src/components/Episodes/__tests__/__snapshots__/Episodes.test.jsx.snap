// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes test 2 1`] = `
<div
  className="tv-show"
/>
`;
